src/Tickets/Emails/Email_Abstract.php

```diff
@@ -391,6 +381,260 @@ public function get_edit_url() {
 	 * @return string
 	 */
 	public function get_id(): string {
-		return static::$id;
+		return $this->id;
+	}
+
+	/**
+	 * Get setting option key.
+	 * 
+	 * @since TBD
+	 * 
+	 * @return string
+	 */
+	public function get_option_key( $option ): string {
+		$template_name = $this->template;
+		return "tec-tickets-emails-{$template_name}-{$option}";
+	}
+
+	/**
+	 * Checks if this email is enabled.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function is_enabled(): bool {
+		$option_key = $this->get_option_key( 'enabled' );
+		return tribe_is_truthy( tribe_get_option( $option_key, true ) );
+	}
+
+	/**
+	 * Get email recipient.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The email recipient.
+	 */
+	public function get_recipient(): string {
+		$option_key = $this->get_option_key( 'recipient' );
+		$recipient = tribe_get_option( $option_key, $this->get_default_recipient() );
+
+		// @todo: Probably we want more data parsed, or maybe move the filters somewhere else as we're always gonna
+
+		/**
+		 * Allow filtering the email recipient globally.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $recipient  The email recipient.
+		 * @param string $id         The email id.
+		 * @param string $template   Template name.
+		 */
+		$recipient = apply_filters( 'tec_tickets_emails_recipient', $recipient, $this->id, $this->template );
+
+		/**
+		 * Allow filtering the email recipient for Completed Order.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $recipient  The email recipient.
+		 * @param string $id         The email id.
+		 * @param string $template   Template name.
+		 */
+		$template_name = $this->template;
+		$recipient = apply_filters( "tec_tickets_emails_{$template_name}_recipient", $recipient, $this->id, $this->template );
```

Comment:
The same thing about the `$template`, we should rely on the $id or something new that stores a unique immutable value.

Reply:
We'll use the new `slug` variable.